fontbe/src/features.rs

```diff
@@ -321,21 +336,29 @@ impl<'a> VariationInfo for FeaVariationInfo<'a> {
         &self,
         values: &HashMap<NormalizedLocation, i16>,
     ) -> Result<(i16, Vec<(VariationRegion, i16)>), Error> {
-        let var_model = &self.static_metadata.variation_model;
-
         // Compute deltas using f64 as 1d point and delta, then ship them home as i16
         let point_seqs: HashMap<_, _> = values
             .iter()
             .map(|(pos, value)| (pos.clone(), vec![*value as f64]))
             .collect();
 
-        // We only support use when the point seq is at a location our variation model supports
-        // TODO: get a model for the location we are asked for so we can support sparseness
-        for loc in point_seqs.keys() {
-            if !var_model.supports(loc) {
-                return Err(Error::NoVariationModel(loc.clone()));
-            }
-        }
+        let locations: HashSet<_> = point_seqs.keys().collect();
+        let global_locations: HashSet<_> =
+            self.static_metadata.variation_model.locations().collect();
+
+        // Try to reuse the global model, or make a new sub-model only with the locations we
+        // are asked for so we can support sparseness
+        let local_model: VariationModel;
+        let var_model = if locations == global_locations {
+            &self.static_metadata.variation_model
+        } else {
+            local_model = VariationModel::new(
+                locations.into_iter().cloned().collect(),
+                self.static_metadata.axes.clone(),
+            )
+            .unwrap();
+            &local_model
+        };
```

Comment:
yeah. If we do consolidate these two implementations of resolve_variable_metric, maybe we could use a struct and let the method take `&mut self` so we can store a cache of prebuilt variation models to reuse when we see the same set of locations, instead of making new ones each time

Reply:
https://github.com/googlefonts/fontc/issues/781